include/deal.II/multigrid/mg_transfer_interface.templates.h

```diff
@@ -0,0 +1,357 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_mg_transfer_inteface_templates_h
+#define dealii_mg_transfer_inteface_templates_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/matrix_free/evaluation_kernels.h>
+
+#include <deal.II/multigrid/mg_transfer_interface.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace
+{
+  /**
+   * Helper class to select the right templated implementation.
+   */
+  class CellTransfer
+  {
+  public:
+    CellTransfer(const unsigned int degree_fine,
+                 const unsigned int degree_coarse)
+      : n(100 * degree_fine + degree_coarse)
+    {}
+
+    template <typename Fu>
+    bool
+    run(Fu &fu)
+    {
+      return do_run_degree<1>(fu) || do_run_degree<2>(fu) ||
```

Comment:
what about degree=0?

Reply:
We can think about `degree=0`. But let's leave that for a follow-up PR.